packages/controller-config/src/resources/dockerhub.ts

```diff
@@ -0,0 +1,135 @@
+/**
+ * Copyright 2021 Opstrace, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { KubeConfig, V1PodSpec } from "@kubernetes/client-node";
+import {
+  Namespaces,
+  ResourceCollection,
+  Secret,
+  Secrets
+} from "@opstrace/kubernetes";
+import { log } from "@opstrace/utils";
+
+export const OPSTRACE_DOCKERHUB_USERNAME = "OPSTRACE_DOCKERHUB_USERNAME";
+export const OPSTRACE_DOCKERHUB_TOKEN = "OPSTRACE_DOCKERHUB_TOKEN";
+export const DOCKERHUB_CREDS_SECRET_NAME = "dockerhub-credentials";
+const DOCKERHUB_CREDS_NAMESPACE = "kube-system";
+
+export function dockerHubCredsSecret(kubeConfig: KubeConfig) {
+  const collection = new ResourceCollection();
+  const username = process.env[OPSTRACE_DOCKERHUB_USERNAME];
+  const token = process.env[OPSTRACE_DOCKERHUB_TOKEN];
+
+  if (!(username && token)) {
+    log.info(
+      "OPSTRACE_DOCKERHUB_USERNAME, OPSTRACE_DOCKERHUB_TOKEN not present, skip image pull secrets"
+    );
+    return collection;
+  }
+
+  const dockerconfig = `{"auths":{"https://index.docker.io/v2/":{"auth":"${Buffer.from(
+    `${username}:${token}`
+  ).toString("base64")}"}}}`;
```

Comment:
Let's info-log that these credentials are now being communicated into the Opstrace instance via a k8s secret. The log line should contain the Docker Hub username, and the first and last character of the token as a bonus; for debuggability.

Reply:
Can we maybe make `const dockerconfig` be a string holding the actual config verbatim?

Can then do the translation of that string into gibberish further below. 

Right now it appears as if we are base64-encoding twice. If that is correct it maybe needs a code comment!

Reply:
simpler, clearer, probably more correct:
```typescript
const jsonstring = `{"auths":{"https://index.docker.io/v2/":{"auth":"${username}:${token}"}}}`
```

and then
```
".dockerconfigjson": Buffer.from(jsonstring).toString("base64")
```

Reply:
:+1: